node-hub/ultralytics-yolo/main.py

```diff
@@ -0,0 +1,158 @@
+    "fire hydrant",
+    "stop sign",
+    "parking meter",
+    "bench",
+    "bird",
+    "cat",
+    "dog",
+    "horse",
+    "sheep",
+    "cow",
+    "elephant",
+    "bear",
+    "zebra",
+    "giraffe",
+    "backpack",
+    "umbrella",
+    "handbag",
+    "tie",
+    "suitcase",
+    "frisbee",
+    "skis",
+    "snowboard",
+    "sports ball",
+    "kite",
+    "baseball bat",
+    "baseball glove",
+    "skateboard",
+    "surfboard",
+    "tennis racket",
+    "bottle",
+    "wine glass",
+    "cup",
+    "fork",
+    "knife",
+    "spoon",
+    "bowl",
+    "banana",
+    "apple",
+    "sandwich",
+    "orange",
+    "broccoli",
+    "carrot",
+    "hot dog",
+    "pizza",
+    "donut",
+    "cake",
+    "chair",
+    "couch",
+    "potted plant",
+    "bed",
+    "dining table",
+    "toilet",
+    "tv",
+    "laptop",
+    "mouse",
+    "remote",
+    "keyboard",
+    "cell phone",
+    "microwave",
+    "oven",
+    "toaster",
+    "sink",
+    "refrigerator",
+    "book",
+    "clock",
+    "vase",
+    "scissors",
+    "teddy bear",
+    "hair drier",
+    "toothbrush",
+]
+
+
+def main():
+    # Handle dynamic nodes, ask for the name of the node in the dataflow, and the same values as the ENV variables.
+    parser = argparse.ArgumentParser(
+        description="UltraLytics YOLO: This node is used to perform object detection using the UltraLytics YOLO model.")
+
+    parser.add_argument("--name", type=str, required=False, help="The name of the node in the dataflow.",
+                        default="ultralytics-yolo")
+    parser.add_argument("--model", type=str, required=False,
+                        help="The name of the model file (e.g. yolov8n.pt).", default="yolov8n.pt")
+
+    args = parser.parse_args()
+
+    model_path = os.getenv("MODEL", args.model)
+
+    model = YOLO(model_path)
+    node = Node(args.name)
+
+    pa.array([])  # initialize pyarrow array
+
+    for event in node:
+        event_type = event["type"]
+
+        if event_type == "INPUT":
+            event_id = event["id"]
+
+            if event_id == "image":
+                arrow_image = event["value"][0]
+                image = {
+                    "width": np.uint32(arrow_image["width"].as_py()),
+                    "height": np.uint32(arrow_image["height"].as_py()),
+                    "channels": np.uint8(arrow_image["channels"].as_py()),
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                }
+
+                frame = image["data"].reshape((image["height"], image["width"], 3))
+
+                frame = frame[:, :, ::-1]  # OpenCV image (BGR to RGB)
+                results = model(frame, verbose=False)  # includes NMS
+
+                bboxes = np.array(results[0].boxes.xyxy.cpu())
+                conf = np.array(results[0].boxes.conf.cpu())
+                labels = np.array(results[0].boxes.cls.cpu())
+
+                names = [LABELS[int(label)] for label in labels]
```

Comment:
Could be good to make the names not hardcoded.

In my previous commit, it looked like this:


```suggestion
                names = [model.names.get(label) for label in labels]
```

Reply:
Yeah it will be better, I missed the names properties in the doc ahah

```Python
class YOLOWorld(Model):
    """YOLO-World object detection model."""

    def __init__(self, model="yolov8s-world.pt", verbose=False) -> None:
        """
        Initializes the YOLOv8-World model with the given pre-trained model file. Supports *.pt and *.yaml formats.

        Args:
            model (str | Path): Path to the pre-trained model. Defaults to 'yolov8s-world.pt'.
        """
        super().__init__(model=model, task="detect", verbose=verbose)

        # Assign default COCO class names when there are no custom names
        if not hasattr(self.model, "names"):
            self.model.names = yaml_load(ROOT / "cfg/datasets/coco8.yaml").get("names")
```